uktrade/data-hub-frontend
# Feature branch/tet 613 refactor investment project

## Description of change

Refactor project investment pages so that the layout is rendered before the resource is loaded. Create new layouts for the investment pages to use so that inline resources can be rendered for better user experience if there are errors.

## Test instructions
These steps use 'Super Awesome Project' as an example investment project but you can use any investment project to test the changes.

To carry out these steps, run the dev environment locally.

1. From the Data Hub homepage, navigate to the [Investments page](http://localhost:3000/investments/projects?page=1&sortby=created_on%3Adesc) 

2. Select an Investment Project to load the **Project details** page e.g. [Super Awesome Project](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/details)

3. You should see the page layout load before the page resources i.e. you should be able to see the page header and left hand navigation whilst the Investment information is loaded (see screen shots below for an example). Loading wheels should appear inline.

4. To test that the page displays error messages inline when an investment cannot be loaded, update the ProjectId to an unrecognisable string
e.g. 

> const ProjectTeam = () => {
> // const { projectId } = useParams()
> const projectId = 'asdf'

5. Repeat the previous steps for the following pages within the investment project:
- [Project team](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/team)
- [Edit client relationship management](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/edit-client-relationship-management)
- [Edit team members](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/edit-team-members)
- [Tasks](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/tasks?sortby=-created_on)
- [Interactions](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/interactions)
- [Propositions](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/propositions?sortby=-created_on&page=1)
- [Evaluations](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/evaluation)
- [Edit history](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/edit-history)
- [Evidence](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/evidence)

## Screenshots

### Before
Appearance of [Project Team page](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/team) with no errors:
<img width="1497" alt="Screenshot 2024-01-31 at 10 08 04" src="https://github.com/uktrade/data-hub-frontend/assets/74198488/87a9ede1-6a1b-4d54-89aa-de98135c268d">

### After
Appearance of [Project Team page](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/team) with no errors:
<img width="1497" alt="Screenshot 2024-01-31 at 10 09 59" src="https://github.com/uktrade/data-hub-frontend/assets/74198488/099bca10-3840-4a97-bdc7-18c2f72bac8f">


### Before
Appearance of [Project Team page](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/team) with errors:
<img width="1497" alt="Screenshot 2024-01-31 at 10 52 40" src="https://github.com/uktrade/data-hub-frontend/assets/74198488/e91839bd-a1f2-4371-86dc-2a8ea7775e43">

### After
Appearance of [Project Team page](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/team) with errors:
<img width="1497" alt="Screenshot 2024-01-31 at 10 44 06" src="https://github.com/uktrade/data-hub-frontend/assets/74198488/22840cdd-3566-4237-8694-808def41e640">


## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'
